w3lib/encoding.py

```diff
@@ -281,27 +281,16 @@ def html_to_unicode(
     >>>
 
     '''
-
-    enc = http_content_type_encoding(content_type_header)
     bom_enc, bom = read_bom(html_body_str)
-    if enc is not None:
-        # remove BOM if it agrees with the encoding
-        if enc == bom_enc:
-            bom = cast(bytes, bom)
-            html_body_str = html_body_str[len(bom) :]
-        elif enc == "utf-16" or enc == "utf-32":
-            # read endianness from BOM, or default to big endian
-            # tools.ietf.org/html/rfc2781 section 4.3
-            if bom_enc is not None and bom_enc.startswith(enc):
-                enc = bom_enc
-                bom = cast(bytes, bom)
-                html_body_str = html_body_str[len(bom) :]
-            else:
-                enc += "-be"
```

Comment:
I'm trying to add tests pertaining to this one but I'm not sure why this is needed.

```
>>> to_unicode(b'\xa3\x00', 'utf-16')
'£'
>>> to_unicode(b'\xa3\x00', 'utf-16-be')
'ꌀ'
>>> to_unicode(b'\xa3\x00', 'utf-16-le')
'�'
>>> to_unicode(b'\xff\xfe\x00\x00\xa3\x00\x00\x00', 'utf-32')
'£'
>>> to_unicode(b'\xff\xfe\x00\x00\xa3\x00\x00\x00', 'utf-32-be')
'��'
>>> to_unicode(b'\xff\xfe\x00\x00\xa3\x00\x00\x00', 'utf-32-le')
'\ufeff£'
```

Reply:
I get the same decoding for `utf-16` and `utf-16-le`:

```
>>> import w3lib
>>> w3lib.__version__
'2.0.1'
>>> from w3lib.encoding import to_unicode
>>> to_unicode(b'\xa3\x00', 'utf-16-le')
'£'
>>> to_unicode(b'\xa3\x00', 'utf-16')
'£'
```

Reply:
@Gallaecio what about the big endian value?

Reply:
That's weird, I now get the same value for both `utf-16` and `utf-16-le` as well. Perhaps I copy-pasted it wrong. 😅

Reply:
The output being the same for those 2 cases (but not for big endian), does your comment still apply?
The old code makes sense to me (i.e. apply big endian if the Content-Type matches the BOM, and the BOM indicates the endianess, which I take the Content-Type cannot or may not always indicate), and it makes sense to me that we do not need it anymore now that the BOM takes precedence.

Reply:
> it makes sense to me that we do not need it anymore now that the BOM takes precedence.

We still need it since the logic of adding "-be" as an encoding suffix only applies when the BOM is not present (or the bom is not the same with the content type headers).

In any case, I just realized that these tests in https://github.com/scrapy/w3lib/blob/master/tests/test_encoding.py#L254-L256 would fail if we do remove such logic. So I guess somehow it's still needed if the BOM isn't there.